fpm/src/fpm/cmd/new.f90

```diff
@@ -2,51 +2,79 @@ module fpm_cmd_new
 
 use fpm_command_line, only : fpm_new_settings
 use fpm_environment, only : run, OS_LINUX, OS_MACOS, OS_WINDOWS
-use fpm_filesystem, only : join_path, exists, basename, mkdir
+use fpm_filesystem, only : join_path, exists, basename, mkdir, is_dir
 use,intrinsic :: iso_fortran_env, only : stderr=>error_unit
 implicit none
 private
 public :: cmd_new
 
 contains
 
-subroutine cmd_new(settings) ! --with-executable F --with-test F '
+subroutine cmd_new(settings)
 type(fpm_new_settings), intent(in) :: settings
 character(len=:),allocatable :: bname          ! baeename of NAME
 character(len=:),allocatable :: message(:)
 character(len=:),allocatable :: littlefile(:)
+character(len=8)             :: date
+
+    call date_and_time(DATE=date)
+
+    if(exists(settings%name) .and. .not.settings%backfill )then
+        write(stderr,'(*(g0,1x))')&
+        & 'ERROR: ',settings%name,'already exists.'
+        write(stderr,'(*(g0,1x))')&
+        & '        perhaps you wanted to add --backfill ?'
+        return
+    elseif(is_dir(settings%name) .and. settings%backfill )then
+        write(*,'(*(g0))')'backfilling ',settings%name
+    elseif(exists(settings%name) )then
+        write(stderr,'(*(g0,1x))')&
+        & 'ERROR: ',settings%name,'already exists and is not a directory.'
+        return
+    else
+        ! make new directory
+        call mkdir(settings%name)
+    endif
 
-    call mkdir(settings%name)      ! make new directory
-    call run('cd '//settings%name) ! change to new directory as a test. System dependent potentially
-    !! NOTE: need some system routines to handle filenames like "." like realpath() or getcwd().
+    ! change to new directory as a test. System dependent potentially
+    call run('cd '//settings%name)
+    !! NOTE: need some system routines to handle filenames like "."
+    !! like realpath() or getcwd().
     bname=basename(settings%name)
 
-    !! weird gfortran bug?? lines truncated to concatenated string length, not 80
-    !! hit some weird gfortran bug when littlefile data was an argument to warnwrite(3f), ok when a variable
+    !! weird gfortran bug?? lines truncated to concatenated string length,
+    !! not 80
+    !! hit some weird gfortran bug when littlefile data was an argument
+    !! to warnwrite(3f), ok when a variable
```

Comment:
Same here, please avoid `!!` unless the lines should be included in the API documentation.

Reply:
Never used FORD. I use ufpp as it supports in-line text blocks for variables and documentation. Is there some reserved comment type other than a plain ! (maybe) that doxygen, FORD, etc. will ignore that should be adapted as a standard for comments that can be flagged as non-standard?  I imagine a bunch of those tools depend on special comments?  For the same reason that !! is special to FORD there are reasons to want special comments to be grep-able or easily identified

Reply:
FORD is currently used to generate the API documentation for the Fortran stdlib, see https://stdlib.fortran-lang.org.